website_form_recaptcha: fix and backward compat

simahawk opened the pull request:
@chienandalu regarding your change from https://github.com/OCA/website/pull/540 can you check it is properly handled? 
I added a specific test for it, still, better to double check :)

I'd also like to use vcrpy to remove most of the mockings in tests, but this is something that can wait.

I added also some backward compat glue to address issues like https://github.com/OCA/apps-store/pull/46
/cc @bizzappdev @pedrobaeza

yajo commented on website_form_recaptcha/models/website_form_recaptcha.py:
```diff
@@ -80,32 +82,56 @@ def validate_response(self, response, remote_ip, website=None):
+    def action_validate(self, req_value, ip_addr):
+        """Backward compatibility for old implementation.
+
+        Prior to API refactoring pre-validation steps
```
Could you specify which API refactoring? Maybe put a link or something. I'm confused if it's Odoo's, OCA's, or Google's.

yajo commented on website_form_recaptcha/tests/test_captcha.py:
```diff
@@ -95,6 +95,43 @@ def test_no_success_raises(self, mk):
+        request = mock.MagicMock()
+        request.g_recaptcha_response = 'all good here'
+        req_values = {}
+        self.assertTrue(self.model_obj.validate_request(request, req_values))
```
In this specific case, you probably need to ensure no calls to validation method are done, since the request is repeated.

It would be a bug if the validation is called when it shouldn't.

yajo commented on website_form_recaptcha/tests/test_captcha.py:
```diff
@@ -95,6 +95,43 @@ def test_no_success_raises(self, mk):
+        # Ensure that w/ proper conditions `validate_response is called`
+        request = mock.MagicMock()
+        request.g_recaptcha_response = None
+        # backwarrd compat skip
```
"Backward"

yajo commented on website_form_recaptcha/tests/test_captcha.py:
```diff
@@ -95,6 +95,43 @@ def test_no_success_raises(self, mk):
+            self.model_obj.action_validate(
+                'captcha_value', '4.5.6.7'
+            )
+            mocked.assert_called_with('captcha_value', '4.5.6.7')
```
In this specific case (different from the above) you are testing a method implementation instead of its results.

This is a good idea for the case explained above, but IMHO not here. The fact that it calls another method doesn't mean it's going to return that value. It would be better to check instead that the method output is expected.

Also, AFAICS, this test should disappear on v12, right? A TODO note will ease the migration.

simahawk commented:
@Yajo you are right on all the points. Addressed. For backward compat tests, to be 100% it behaves as the new api I replicated all the main tests. 
Also, I added a deprecation warning for the old method.
